### PR TITLE
Telestion Core `v0.7`

### PR DESCRIPTION
**Pull request by fussel178**
### Summary 

Update extension to Telestion Core `v0.7`.

### Details 

n/a

### Additional information 

n/a

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### CODE REVIEW

File: src/main/java/de/wuespace/telestion/extension/mongodb/PeriodicDataAggregator.java [hunk 1]
```diff
@@ -1,180 +1,117 @@
...
+	private void databaseRequest(Long timerId) {
+		logger.debug("Executing aggregation request...");
+
+		// build query to aggregate data newer than last published dataset
+		var dateQuery = new JsonObject()
+				.put("datetime", new JsonObject().put("$gt", new JsonObject().put("$date", timeOfLastDataSet)));
+		var dbRequest = new DbRequest(getConfig().collection(), dateQuery.toString(), getConfig().field());
+
+		// request on database verticle
+		this.<JsonObject>request(IN_AGGREGATE, dbRequest)
```
Comment (pklaschka) [inline src/main/java/de/wuespace/telestion/extension/mongodb/PeriodicDataAggregator.java:60]: Oh, how I hate Java's generic syntax, but nothing we can do about that 🙈

**Review by pklaschka — APPROVED** [review]
LGTM. Thank you for your contribution 🎉 🙂

File: src/main/java/de/wuespace/telestion/extension/mongodb/DataService.java [hunk 2]
```diff
@@ -72,106 +48,55 @@ public Configuration getConfig() {
...
-					resultHandler.handle(Future.failedFuture(res.cause().getMessage()));
-					return;
-				}
-				dataOperation.data().put("data", res.result());
+	private void dataRequestDispatcher(DataRequest request, Message<JsonObject> message) {
+		var latestData = fetchLatestData(request.collection(), request.query());
+
+		// manipulate latest data if request is more specific
+		if (!request.collection().isEmpty()) {
+			latestData = latestData.compose(result -> {
```
Comment (fussel178) [inline src/main/java/de/wuespace/telestion/extension/mongodb/DataService.java:59]: Here, the returned future from `compose` is from interest, not the actual latest data.

**Review by fussel178 — COMMENTED** [review]
Sorry for the inconvenience. I've created a little error in the `DataService` verticle. :see_no_evil:

**Review by pklaschka — APPROVED** [review]
LGTM. Thank you for your contribution 🎉 🙂

**Pull request merged**